lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,399 @@
+                "file": str(self.input_electron_dl2),
+                "target_spectrum": IRFDOC_ELECTRON_SPECTRUM,
+            }
+
+            Provenance().add_input_file(self.input_proton_dl2)
+            Provenance().add_input_file(self.input_electron_dl2)
+
+        self.provenance_log = self.output_irf_file.parent / (
+            self.name + ".provenance.log"
+        )
+
+    def start(self):
+
+        for particle_type, p in self.mc_particle.items():
+            self.log.info(f"Simulated {particle_type.title()} Events:")
+            p["events"], p["simulation_info"] = read_mc_dl2_to_QTable(p["file"])
+
+            if p["simulation_info"].viewcone.value == 0.0:
+                p["mc_type"] = "point_like"
+            else:
+                p["mc_type"] = "diffuse"
+                # For diffuse gamma using Proton Spectra for calculating event weights
+                if particle_type == "gamma":
+                    if not self.point_like:
+                        p["target_spectrum"] = IRFDOC_PROTON_SPECTRUM
+                        self.log.debug(
+                            "Proton spectrum used as target spectrum"
+                            " for MC diffuse gamma"
+                        )
+                    else:
+                        raise ToolConfigurationError(
+                            "Diffuse MC gamma cannot be used for generating "
+                            "point_like IRFs. Use appropriate MC and IRF type."
+                        )
+
+            self.log.debug(f"Simulated {p['mc_type']} {particle_type.title()} Events:")
+
+            p["simulated_spectrum"] = PowerLaw.from_simulation(
+                p["simulation_info"], 50 * u.hour
+            )
+            p["events"]["weight"] = calculate_event_weights(
+                p["events"]["true_energy"],
+                p["target_spectrum"],
+                p["simulated_spectrum"],
+            )
+
+            for prefix in ("true", "reco"):
+                k = f"{prefix}_source_fov_offset"
+                p["events"][k] = calculate_source_fov_offset(p["events"], prefix=prefix)
+            # calculate theta / distance between reco and assumed source position
+            p["events"]["theta"] = calculate_theta(
+                p["events"],
+                assumed_source_az=p["events"]["true_az"],
+                assumed_source_alt=p["events"]["true_alt"],
+            )
+            self.log.debug(p["simulation_info"])
+
+        gammas = self.mc_particle["gamma"]["events"]
+
+        self.log.info(f"Using fixed G/H cut of {self.data_sel.fixed_gh_cut}")
+
+        gammas = self.data_sel.filter_cut(gammas)
+        gammas = self.data_sel.allowed_tels_filter(gammas)
+        gammas = self.data_sel.gh_cut(gammas)
+
+        # point_like = True for point like IRFs, False for Full Enclosure IRFs
+        if self.point_like:
+            gammas = self.data_sel.theta_cut(gammas)
+            gammas = self.data_sel.true_src_fov_offset_cut(gammas)
```

Comment:
Why this true_source_offset_cut and why only for point like?

Reply:
The true_source_fov_offset_cut is for applying another angular cut when we want point-like IRF

Reply:
But why? For what is this cut needed? If you have point-like simulations, this cut will not do anything and for diffuse gammas, it will result in the wrong effective area at high fov offsets.

Where does this cut come from?

Reply:
Ah yes! I had misunderstood this parameter for a long time and foolishly tried to apply a cut!